server/website/website/views.py

```diff
@@ -532,13 +532,16 @@ def handle_result_files(session, files):
         response = chain(aggregate_target_results.s(result.pk),
                          map_workload.s(),
                          configuration_recommendation.s()).apply_async()
-    elif session.algorithm == AlgorithmType.ALGORITHM1:
+    elif session.algorithm == AlgorithmType.DDPG:
+        response = chain(train_ddpg.s(result.pk),
+                         configuration_recommendation_ddpg.s()).apply_async()
+    elif session.algorithm == AlgorithmType.DNN:
         pass
-    elif session.algorithm == AlgorithmType.ALGORITHM2:
-        pass
-    elif session.algorithm == AlgorithmType.ALGORITHM3:
-        pass
-    taskmeta_ids = [response.parent.parent.id, response.parent.id, response.id]
+    taskmeta_ids = []
+    current_task = response
+    while current_task:
+        taskmeta_ids.append(current_task.id)
+        current_task = current_task.parent
```

Comment:
taskmeta_ids = [response.parent.parent.id, response.parent.id, response.id]
instead of taskmeta_ids = [response.id, response.parent.id,response.parent.parent.id]

After this change, The GPR recommended result is shown on the Preprocess part

Reply:
Will change taskmeta_ids.append(current_task.id) to taskmeta_ids.insert(0, current_task.id)

Reply:
Nice catch.